content/blog/2025-05-08-go-client-GA.md

```diff
@@ -0,0 +1,503 @@
+
+```go
+cmd.Set("key1","val1")
+cmd.Get("key1")
+```
+
+**Execute Transaction:**
+
+ Run all queued commands atomically
+ After queueing the commands, transaction can be executed with `tx.Exec()`
+ All the queued commands will be executed and returns an response array which consists of the commands result.
+
+## Standalone Mode Transaction
+
+```go
+    // Standalone Client Transaction
+    config := api.NewGlideClientConfiguration().
+        WithAddress(&api.NodeAddress{Host: "localhost", Port: 6379})
+
+    client, err := api.NewGlideClient(config)
+    if err != nil {
+        log.Fatal("error connecting to database: ", err)
+    }
+    defer client.Close()
+
+    tx := api.NewTransaction(client)
+    cmd := tx.GlideClient
+
+    // Queue multiple commands
+    cmd.Set("user:profile", "John Doe")
+    cmd.Set("user:email", "john@example.com")
+    cmd.SAdd("users", "john")
+
+    // Execute transaction
+    results, err := tx.Exec()
+    if err != nil {
+        log.Printf("Transaction failed: %v", err)
+        return
+    }
+
+    // Process results
+    for _, result := range results {
+        fmt.Println(result)
+    }
+```
+
+## Cluster mode Transaction
+
+To create a cluster transaction, initialize the cluster transaction object using `api.NewClusterTransaction(clusterClient)` to create a new cluster transaction with an established cluster client as an argument.
+`cmd := tx.GlideClusterClient` grants access to a transaction-scoped cluster commands interface, allowing you to construct a sequence of operations to be executed.
+
+```go
+    // Cluster Client Transaction
+    clusterConfig := api.NewGlideClusterClientConfiguration().
+        WithAddress(&api.NodeAddress{Host: "cluster1", Port: 7007}).
+        WithAddress(&api.NodeAddress{Host: "cluster2", Port: 7008}).
+        WithAddress(&api.NodeAddress{Host: "cluster3", Port: 7009})
+
+    clusterClient, err := api.NewGlideClusterClient(clusterConfig)
+    if err != nil {
+        log.Fatal("error connecting to cluster: ", err)
+    }
+    defer clusterClient.Close()
+
+    clusterTx := api.NewClusterTransaction(clusterClient)
+    cmd := clusterTx.GlideClusterClient
+
+    // Define routing option
+    var simpleRoute config.Route = config.RandomRoute
+
+    // Create ping options with specific routing
+    pingOpts := options.ClusterPingOptions{
+        PingOptions: &options.PingOptions{
+            Message: "Hello Valkey Cluster",
+        },
+        RouteOption: &options.RouteOption{
+            Route: simpleRoute,
+        },
+    }
+
+    // Queue commands with routing
+    cmd.PingWithOptions(pingOpts)
+    // Execute transaction
+    results, err := clusterTx.Exec()
+    if err != nil {
+        log.Printf("Cluster transaction failed: %v", err)
+        return
+    }
+
+    // Process results
+    for _, result := range results {
+        fmt.Println(result)
+    }
+```
+
+In cluster transactions, the routing option does not enable multi-node transaction execution.
```

Comment:
How important is this? It gets lost below the example.

In general, this type of writing isn't great for "does not" messages - your announcing what something is not what it can't do.